include/openssl/core.h

```diff
@@ -25,6 +25,11 @@ extern "C" {
  * to communicate data between them.
  */
 
+/* Opaque handles to be used with core upcall functions */
+typedef void *OSSL_CORE_HANDLE;
+typedef void *OPENSSL_CORE_CTX;
+typedef void *OSSL_CORE_BIO;
```

Comment:
Likewise for these, please avoid building in the pointer indicator (`*`) in the typedef.  You're obviously also confusing yourself, considering that you're using `OPENSSL_CORE_CTX *` and `OSSL_CORE_BIO *` all over the place :wink: (even though casting "fixes" it, it's still potentially dangerous practice)

Reply:
Fixed